tests/integration/ambient/baseline_test.go

```diff
@@ -1441,13 +1606,109 @@ spec:
 							from.CallOrFail(t, echo.CallOptions{
 								Address: "111.111.222.222",
 								Port:    to.PortForName("http"),
+								// we want to make sure the workload selector labels are honored, so we assert the response contains either uncaptured-v1 or uncaptured-v2
+								// sample response:
+								//
+								// ServiceVersion=v1
+								// ServicePort=8080
+								// Host=serviceentry.istio.io
+								// URL=/any/path
+								// Cluster=cluster-0
+								// IstioVersion=
+								// Method=GET
+								// Proto=HTTP/1.1
+								// IP=10.244.2.20
+								// Alpn=
+								// RequestHeader=Accept:*/*
+								// RequestHeader=User-Agent:curl/7.81.0
+								// Hostname=uncaptured-v1-868c9b59b5-rxvfq
+								Check: check.BodyContains(`Hostname=uncaptured-v`),
 							})
 						})
 				})
 			}
 		})
 }
 
+// This was successfully smoke tested locally by looking at ztunnel config dump
+// to query the auto-assigned VIP behind the hostname.
+func TestServiceEntryDNSWithAutoAssign(t *testing.T) {
+	framework.NewTest(t).
+		Features("traffic.ambient").
+		Run(func(t framework.TestContext) {
+			t.Skip("this will work if we set DNS_AUTO_ALLOCATE=true and once we have https://github.com/istio/ztunnel/pull/536")
+			yaml := `apiVersion: networking.istio.io/v1beta1
+kind: ServiceEntry
+metadata:
+  name: test-service-entry
+spec:
+  hosts:
+  - serviceentry.istio.io
+  ports:
+  - name: http
+    number: 80
+    protocol: HTTP
+    targetPort: 8080
+  location: MESH_EXTERNAL
+  resolution: STATIC # not honored for now; everything is static
+  workloadSelector:
+    labels:
+      app: uncaptured` // cannot select pods captured in ambient mesh; IPs are unique per network
+			svcs := apps.All
+			for _, svc := range svcs {
+				if svc.Config().IsUncaptured() || svc.Config().HasSidecar() {
+					continue
+				}
+				if err := t.ConfigIstio().YAML(svc.NamespaceName(), yaml).Apply(apply.NoCleanup); err != nil {
+					t.Fatal(err)
+				}
+				t.NewSubTestf("%v to ServiceEntry", svc.Config().Service).Run(func(t framework.TestContext) {
+					svc.CallOrFail(t, echo.CallOptions{
+						Address: "serviceentry.istio.io",
+						Port:    echo.Port{Name: "http", ServicePort: 80},
+						Scheme:  scheme.HTTP,
+						HTTP: echo.HTTP{
+							Path: "/any/path",
+						},
+						// we want to make sure the workload selector labels are honored, so we assert the response contains either uncaptured-v1 or uncaptured-v2
+						// sample response:
+						//
+						// ServiceVersion=v1
+						// ServicePort=8080
+						// Host=serviceentry.istio.io
+						// URL=/any/path
+						// Cluster=cluster-0
+						// IstioVersion=
+						// Method=GET
+						// Proto=HTTP/1.1
+						// IP=10.244.2.20
+						// Alpn=
+						// RequestHeader=Accept:*/*
+						// RequestHeader=User-Agent:curl/7.81.0
+						// Hostname=uncaptured-v1-868c9b59b5-rxvfq
+						Check: check.BodyContains(`Hostname=uncaptured-v`),
```

Comment:
nit; stronger assertion to have both v1 +v2 checks serially to ensure we are load balancing

Reply:
addressed